### ISSUE TITLE
-Wlogical-op warning in wallet/scriptpubkeyman.cpp when building current master

### ISSUE THREAD

**Issue by kristapsk** [bitcoin/bitcoin#19912]
```
wallet/scriptpubkeyman.cpp: In member function ‘virtual bool LegacyScriptPubKeyMan::Upgrade(int, bilingual_str&)’:
wallet/scriptpubkeyman.cpp:455:55: warning: logical ‘and’ applied to non-boolean constant [-Wlogical-op]
  455 |     if (m_storage.CanSupportFeature(FEATURE_HD_SPLIT) && CHDChain::VERSION_HD_CHAIN_SPLIT) {
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  CXX      wallet/libbitcoin_wallet_a-wallet.o
```
```
$ gcc --version
gcc (Gentoo 9.2.0-r2 p3) 9.2.0
```
Related - #19015.

**Comment by practicalswift** [comment]
Thanks for reporting! Very nice to see the recently enabled `-Wlogical-op` pay off. Time for `-Werror=logical-op`?

**Comment by achow101** [comment]
#18836 fixes this.

**Comment by practicalswift** [comment]
@achow101 Thanks! I guess #18836 will need additional review time given its size/scope: would it make sense to patch this separately to get the fix in sooner?

**Comment by maskoficarus** [comment]
I've added a standalone fix in #19986 to get this resolved while #18836 gets sorted out.

**Comment by hebasto** [comment]
gcc 10.2.1 (Fedora 32) -- no warnings
gcc 9.3.0 (Linux Mint 20 / Ubuntu Focal) -- a warning is emitted
